src/Sylius/Bundle/ApiBundle/Applicator/OrderStateMachineTransitionApplicator.php

```diff
@@ -39,7 +40,11 @@ public function __construct(private StateMachineFactoryInterface|StateMachineInt
 
     public function cancel(OrderInterface $data): OrderInterface
     {
-        $this->applyTransition($data, OrderTransitions::TRANSITION_CANCEL);
+        try {
+            $this->applyTransition($data, OrderTransitions::TRANSITION_CANCEL);
+        } catch (\Exception) {
```

Comment:
I believe we should catch the `Sylius\Abstraction\StateMachine\Exception\StateMachineExecutionException`. Also, in our `Sylius\Abstraction\StateMachine\CompositeStateMachine`, there are annotations that do not match the interface (I believe, the implementation inherits the docblock from the interface). It would be nice to clean this up.